nolabs/features/drug_discovery/predict_diffdock_docking.py

```diff
@@ -0,0 +1,125 @@
+            confidence = ligand_result.confidence
+            if confidence:
+                minimised_affinity = ligand_result.minimized_affinity
+                scored_affinity = ligand_result.scored_affinity
+                ligand_file_name = ligand_result.sdf_file_name
+                ligand_sdf = ligand_result.sdf_content
+                result_data = DiffDockDockingResultData(confidence=confidence,
+                                                        scored_affinity=scored_affinity,
+                                                        minimized_affinity=minimised_affinity,
+                                                        predicted_sdf_file_name=ligand_file_name,
+                                                        predicted_sdf_contents=ligand_sdf,
+                                                        predicted_pdb=predicted_pdb)
+
+                self._result_file_management.store_diffdock_result_data(experiment_id,
+                                                                        target_id,
+                                                                        ligand_id,
+                                                                        job_id,
+                                                                        result_data)
+
+                predicted_ligands_list.append(DiffDockLigandMetaData(target_id=target_id.value,
+                                                                     ligand_id=ligand_id.value,
+                                                                     job_id=job_id.value,
+                                                                     scored_affinity=result_data.scored_affinity,
+                                                                     minimized_affinity=result_data.minimized_affinity,
+                                                                     confidence=result_data.confidence,
+                                                                     predicted_ligand_file_name=result_data.predicted_sdf_file_name))
+
+        return RunDiffDockDockingJobResponse(predicted_pdb=predicted_pdb,
+                                             predicted_ligands=predicted_ligands_list)
+
+    def get_folding(self, experiment_id: ExperimentId, target_id: TargetId, job_id: JobId, folding_method: str) -> str:
+        if not self._target_file_management.get_pdb_contents(experiment_id, target_id, folding_method):
+            _, sequence, _ = self._target_file_management.get_target_data(experiment_id, target_id)
+
+            host = None
+
+            if folding_method == "esmfold_light":
+                host = self._settings.esmfold_light_host
+            else:
+                host = self._settings.esmfold_host
+
+            configuration = FoldingConfiguration(
+                host=host,
+            )
+            with FoldingApiClient(configuration=configuration) as client:
+                api_instance = FoldingDefaultApi(client)
```

Comment:
api_instance = esmfold_api_instance_factory(settings: Settings)

Reply:
I moved with the other approach, where instead of a factory I simply use methods (see latest commit). This is more useful since some docking instance interfaces will differ (and differ now) in terms of call methods etc. As an example, call and configuration for RosettaFold/Alphafold will include other parameters such as number of recycles etc. and the naming of the post method will be different too